lib/Doctrine/ORM/Query/Lexer.php

```diff
@@ -126,10 +128,11 @@ public function __construct($input)
     protected function getCatchablePatterns()
     {
         return array(
-            '[a-z_\\\][a-z0-9_\:\\\]*[a-z0-9_]{1}',
-            '(?:[0-9]+(?:[\.][0-9]+)*)(?:e[+-]?[0-9]+)?',
-            "'(?:[^']|'')*'",
-            '\?[0-9]*|:[a-z_][a-z0-9_]*'
+            '[a-z_][a-z0-9_]*\:[a-z_][a-z0-9_]*', // aliased name
```

Comment:
Ok, let me try to explain.

Previously, `user`, `Alias:Class` or `Some\Class\Name` were all considered "identifiers". This is in the `'[a-z_\\\][a-z0-9_\:\\\]*[a-z0-9_]{1}'` regexp.

Now there are several parts in the DQL grammar where these `identifiers` were accepted, e. g. `AbstractSchemaName`, `IdentificationVariable` or `AliasIdentificationVariable`. So was possible to use a FQCN the the `IdentificationVariable` and you would not get parser complaints (only other failures further downstream).

At first, I tried to add the `\` and `:` as tokens and then parse `AbstractSchemaName` as a combination of repeated `identifier`, `\` and `:`. In principle, that worked, with the only problem that for example `FROM \Some\ Class` would *also* be accepted. This is due to the way the tokenizer works: It would deliver `FROM`, `\`, `Some`(identifier), `\`, `Class` (identifier). 

So, the only solution I saw was to treat `identifier`, `aliased name` and `fully qualified name` as different tokens at the Lexer level already.

Does that make sense?